devito/ir/iet/scheduler.py

```diff
@@ -65,7 +65,11 @@ def iet_make(stree):
                     # Apart from SteppingDimension, no other type of Dimension
                     # requires generation of uindices
                     continue
-                modulo = len(offs)
+                # Get the modul span. First at 0 (no offset) if not present then
+                # get full offset
+                offs.update({0})
```

Comment:
this might become unnecessary in #594 

look at this line: https://github.com/opesci/devito/pull/594/files#diff-18b194953e1480c7b66bc0641f9c6b9eR82

the extent of the modulo iteration is now dictated by the size of the TimeFunction buffer, not by the stencil indices.

Reply:
Not sure, this litteraly just look at the maximum offsets around the current point, which may be bigger than the `time_order` if for example you take some weird time derivative

Reply:
I think the modulo should be dictated by the extent of the time buffer, not by the offsets used within a particular Operator. Otherwise, you might even generate out-of-bound accesses (and thus segfaults).